rust-runtime/aws-smithy-http-server/src/routing/routers/mod.rs

```diff
@@ -0,0 +1,144 @@
+use std::{
+    fmt::Debug,
+    future::{ready, Future, Ready},
+    marker::PhantomData,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use futures_util::future::Either;
+use http::Response;
+use tower::{util::Oneshot, Service, ServiceExt};
+
+pub mod aws_json;
+// pub mod merged;
+pub mod rest;
+
+use crate::{
+    body::{empty, BoxBody},
+    response::IntoResponse,
+};
+
+const UNKNOWN_OPERATION_EXCEPTION: &str = "UnknownOperationException";
+
+fn method_disallowed() -> http::Response<BoxBody> {
+    http::Response::builder()
+        .status(http::StatusCode::METHOD_NOT_ALLOWED)
+        .body(empty())
+        .expect("valid HTTP response")
```

Comment:
```suggestion
        .expect("invalid HTTP response")
```

Like in the other places.

I'd also be consistent with the other places where we know that if it panics, it's a bug on us, and direct the user to file a report. See https://github.com/awslabs/smithy-rs/blob/a1f2e3edeb0f639029e30da6c1b090ee492bdb3d/rust-runtime/aws-smithy-http-server/src/runtime_error.rs#L92-L92

Reply:
Actually, I'm also thinking whether in these places we can dispense with the panic by doing:

```rust
let mut response = Response::new(empty()); // Or `Response::default()`.
*response.status_mut() = StatusCode::METHOD_NOT_ALLOWED;
```

Reply:
Switched the `method_disallowed` builder to the `Response` mutation. In the cases where headers are set we'd be switching out the explicit `.expect` for a hidden panic in `HeaderValue::from_static`.